File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        //우선순위 큐 사용해서 모든 음식 넣기
        //두개씩 꺼내서 계산하고 K 보다 낮으면 다시 넣기
        //이때 처음 꺼낸 음식이 K보다 크면 break하고 섞은 횟수 return
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        
        for(int i=0; i<scoville.length; i++){
            pq.add(scoville[i]);
        }
        
        int cnt = 0;
        while(true){
            if(pq.size() < 2 && pq.peek() < K) return -1;
            
            int first = pq.poll();
            if(first >= K) return cnt;
            int second = pq.poll();
            
            int newFood = first + (second * 2);
            
            cnt++;
            pq.add(newFood);
        }
    }
}

File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq
def solution(scoville, K):
    #우선순위 큐를 스코빌기준 오름차순 정렬
    #while 사용해서 무한반복
    #처음 뽑은 음식의 스코빌 지수가 K이상이면 while 종료
    #음식 섞을 때마다 섞은 횟수 증가
    pq = []
    cnt = 0
    
    for i in range(len(scoville)):
        heapq.heappush(pq, scoville[i])
        
    while True:
        first = heapq.heappop(pq)
        
        if first >= K:
            break;
        
        if pq:
            second = heapq.heappop(pq)
        else:
            return -1
        
        if first == 0 and second == 0:
            return -1
        
        newFood = first + (second * 2)
        heapq.heappush(pq, newFood)
        cnt += 1
        
    return cnt
        
    
    